fragmenter/tests/test_fragmenter.py

```diff
@@ -18,33 +18,35 @@ def test_fragmenter_imported():
 class TestFragment(unittest.TestCase):
 
     @unittest.skipUnless(has_openeye, "Cannot test without openeye")
-    def test_generate_fragments(self):
-        """ generate fragments regression test"""
+    def test_canoicalization_details(self):
+        """ test canonicalization detail"""
         input_smi = get_fn('butane.smi')
         fragments = fragmenter.generate_fragments(inputf=input_smi)
 
         provenance = fragments['provenance']
         canon_detail = provenance['canonicalization_details']
-        self.assertTrue(canon_detail['AtomMaps'])
-        self.assertTrue(canon_detail['AtomStereo'])
-        self.assertTrue(canon_detail['BondStereo'])
-        self.assertTrue(canon_detail['Canonical'])
-        self.assertTrue(canon_detail['Isotopes'])
-        self.assertTrue(canon_detail['RGroups'])
-        self.assertTrue(canon_detail['ISOMERIC'])
-        self.assertFalse(canon_detail['DEFAULT'])
-
-        with self.assertRaises(Warning):
-            fragmenter.generate_fragments(input_smi, OESMILESFlag='DEFAULT')
-
-        fragments = fragmenter.generate_fragments(input_smi, strict_stereo=False, OESMILESFlag='DEFAULT')
-        provenance = fragments['provenance']
-        canon_detail = provenance['canonicalization_details']
-        self.assertTrue(canon_detail['AtomMaps'])
-        self.assertFalse(canon_detail['AtomStereo'])
-        self.assertFalse(canon_detail['BondStereo'])
-        self.assertTrue(canon_detail['Canonical'])
-        self.assertFalse(canon_detail['Isotopes'])
-        self.assertTrue(canon_detail['RGroups'])
-        self.assertFalse(canon_detail['ISOMERIC'])
-        self.assertTrue(canon_detail['DEFAULT'])
+        isomeric_smiles = canon_detail['canonical_isomeric_SMILES']
+        for flag in isomeric_smiles['Flags_set_to_True']:
```

Comment:
Don't you also need to explicit hydrogens be included?

Reply:
I wonder if we shouldn't have a separate small package that returns a `dict` of the "canonical metadata" we want to include in these JSON files given an `OEMol` as input. That way, we can update the canonical information we store everywhere by just updating that one package.

Reply:
There are 3 SMILES for each fragment:
1) Canonical isomeric SMILES (without explicit hydrogens)
2) Canonical SMILES (without explicit hydrogens) 
3) explicit hydrogen tagged SMARTS - This is where the explicit hydrogens are. The tags are the order the atoms should be in. The SMILES used to generate this tagged SMARTS is canonical isomeric.